cxx4opencl/address_spaces.txt

```diff
@@ -211,44 +291,58 @@ __kernel void foo() {
   [&] () mutable __private {} ();
   [&] () __private mutable {} (); // error: mutable specifier should precede address space.
 }
-----------
-
+------------
 
 ==== Implicit special members
 
 The prototype for implicit special members (default, copy or move constructor,
-copy or move assignment, destructor) has the generic address space for an implicit
+copy or move assignment, destructor) has the default address space for an implicit
 object pointer and reference parameters (see also
-<<addrspace-member-function-qualifiers, _Member function qualifier_>>).
+<<addrspace-member-function-qualifiers, _Member function qualifier_>>). This default
+address space is generic if it is supported or private otherwise.
 
 [source,cpp]
-----------
+------------
 class C {
   // Has the following implicitly defined member functions.
 
-  // void C(); /* implicit 'this' parameter is a pointer to */
-               /* object in generic address space. */
+  // C(); /* implicit 'this' parameter is a pointer to */
+          /* object in generic address space if supported */
+          /* in private address space otherwise. */
 
-  // void C(const C & par); /* 'this'/'par' is a pointer/reference to */
-                            /*  object in generic address space. */
+  // C(const C & par); /* 'this'/'par' is a pointer/reference to */
+                       /*  object in generic address space */
+                       /*  if supported */
+                       /* in private address space otherwise. */
 
-  // void C(C && par); /* 'this'/'par' is a pointer/r-val reference to */
-                       /* object in generic address space. */
+  // C(C && par); /* 'this'/'par' is a pointer/r-val reference to */
+                  /* object in generic address space if supported */
+                  /* in private address space otherwise. */
 
-  // operator= C &(const C & par); /* 'this'/'par'/return value is */
+  // C & operator=(const C & par); /* 'this'/'par'/return value is */
                                    /* a pointer/reference/reference to */
-                                   /* object in generic address space. */
+                                   /* object in generic address space */
+                                   /* if supported */ 
+                                   /* in private address space otherwise. */
 
-  // operator= C &(C && par)'; /* 'this'/'par'/return value is */
+  // C & operator=(C && par)'; /* 'this'/'par'/return value is */
                                /* a pointer/r-val reference/reference to */
                                /* object in generic address space. */
+                               /* if supported */ 
+                               /* in private address space otherwise. */
```

Comment:
Given all this are comments anyway, it would be more readable to use the full line. For example:
```cpp
  // C(); /* implicit 'this' parameter is a pointer to */
          /* object in generic address space if supported */
          /* in private address space otherwise. */
```
would be
```cpp
  // C();
  // implicit 'this' parameter is a pointer to object in generic address space 
  // if supported in private address space otherwise.
```

Reply:
In this document, we break them up because they didn't get visualized properly in the PDF.

However, if we are to do this perhaps it has to be done everywhere? And then it might be better to create separate PR?

Reply:
I have built the pdf and checked and indeed long lines are not displayed nicely in the code examples... so for the time being we need to continue handling the line breaks manually... until we find some automatic solution.... They do look fine in XML because there is a scroll bar in the browsers.